packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -0,0 +1,264 @@
+<template>
+	<main>
+		<header>
+			<section class="header">
+				<!-- <h4>{{ name }}</h4> -->
+				<section class="name">
+					<InputText v-model="codeName" class="name-input" />
+				</section>
+				<section class="buttons">
+					<FileUpload
+						name="demo[]"
+						:customUpload="true"
+						@uploader="onFileOpen"
+						mode="basic"
+						auto
+						chooseLabel="Load file"
+						class="p-button-secondary"
+					/>
+					<Button label="Save code" @click="saveCode" />
+					<Button label="Convert code to model" @click="extractModel" />
+				</section>
+			</section>
+		</header>
+		<v-ace-editor
+			v-model:value="codeText"
+			@init="initialize"
+			lang="python"
+			theme="chrome"
+			style="height: 100%; width: 100%"
+			class="ace-editor"
+		/>
+		<Teleport to="body">
+			<tera-modal
+				v-if="isModalVisible && model"
+				class="modal"
+				@modal-mask-clicked="isModalVisible = false"
+				@modal-enter-press="isModalVisible = false"
+			>
+				<template #header>
+					<InputText v-model="modelName" class="name-input" />
+				</template>
+				<template #default>
```

Comment:
This is where you can add a `name` and `description` field. or even better re-use the create a new model modal.